src/plugins/visualizations/public/vis_types/vis_type_alias_registry.ts

```diff
@@ -79,4 +76,7 @@ export const visTypeAliasRegistry: VisTypeAliasRegistry = {
     }
     registry.push(newVisTypeAlias);
   },
+  remove: (visTypeAliasName) => {
+    registry = registry.filter((visTypeAlias) => visTypeAlias.name !== visTypeAliasName);
```

Comment:
This will leave us with a race condition that a plugin first needs to register an alias before we can remove it, and if we happen to change the order of plugin initializations this might go wrong. I suggest we're doing a pattern similar to the visTypeRegistry here, where in case `visTypeAliasName` does not yet exist in the registry we store it in a temporary `discardOnRegister` array, and in the `add` method checks if the alias type it registers is in this list, and if so just skip registration and remove it from the `discardOnRegister` array. That way we're safe no matter in which way the initialization happens.

Reply:
I assumed that because I register the Alias on the setup but unregister it to the start it will eliminate the race conditions. I see your point, will do the change